### ISSUE TITLE
Pull latest changes for sample repositories

### ISSUE THREAD

**Issue by bamurtaugh** [microsoft/vscode-remote-release#4606]
- VSCode Version: latest Insiders and Stable
Version: 1.54.1 (system setup)
Commit: f30a9b73e8ffc278e71575118b6bf568f04587c8
Date: 2021-03-04T22:38:31.419Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042
- Local OS Version: Windows 10
- Remote Extension/Connection Type: Remote - Containers

Steps to Reproduce:

**Remote-Containers: Try a sample** and choose the Python sample.

We updated the remote-try-* samples in January to use dynamic port forwarding and reference GitHub Codespaces in their readme's. It looks like the samples are overall working as intended (i.e. when I do **Remote-Containers: Try a sample**, they pull the latest updates), except for https://github.com/microsoft/vscode-remote-try-python? 

As you can see, the version it pulled doesn't have the latest updates (i.e. no mention of Codespaces in the readme):

![image](https://user-images.githubusercontent.com/25310137/110154674-5e454b80-7d99-11eb-9abf-e82337c30e86.png)

**Comment by bamurtaugh** [comment]
I also get this error when opening certain try-* samples via the Try a Sample command in Remote-Containers. I got it when I tried opening the Python sample for the first time, and now I just got it when trying to open the Go sample:

![image](https://user-images.githubusercontent.com/25310137/110161313-08c16c80-7da2-11eb-9801-a29305078e98.png)

```
[16 ms] Remote-Containers 0.163.0 in VS Code 1.54.1 (f30a9b73e8ffc278e71575118b6bf568f04587c8).
[406 ms] Start: Resolving Remote
[410 ms] Start: Check Docker is running
[410 ms] Start: Run: docker version --format {{.Server.APIVersion}}
[803 ms] Server API version: 1.41
[804 ms] Start: Run: docker volume ls -q
[1346 ms] Start: Run: docker ps -q -a --filter label=vsch.local.repository=https://github.com/Microsoft/vscode-remote-try-go --filter label=vsch.local.repository.volume=vsc-remote-sample --filter label=vsch.local.repository.folder=vscode-remote-try-go
[1821 ms] Start: Run: docker inspect --type container 93ab877cb144
[2362 ms] Start: Starting container
[2362 ms] Start: Run: docker start 93ab877cb144fbfeb08936a64f46fff346b3e7e8ecdc4f7ee1256e12a823bf0a
[3355 ms] Start: Run: docker ps -q -a --filter label=vsch.local.repository=https://github.com/Microsoft/vscode-remote-try-go --filter label=vsch.local.repository.volume=vsc-remote-sample --filter label=vsch.local.repository.folder=vscode-remote-try-go --filter label=vsch.quality=stable
[3869 ms] Dev container not found.
[3876 ms] Error: Dev container not found.
        at bf (c:\Users\brmurtau\.vscode\extensions\ms-vscode-remote.remote-containers-0.163.0\dist\extension\extension.js:186:13667)
        at Tg (c:\Users\brmurtau\.vscode\extensions\ms-vscode-remote.remote-containers-0.163.0\dist\extension\extension.js:186:11279)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
        at async pg (c:\Users\brmurtau\.vscode\extensions\ms-vscode-remote.remote-containers-0.163.0\dist\extension\extension.js:186:7020)
```

**Comment by chrmarti** [comment]
Copied second issue to #4625.

Is the Python repository in your Docker volume on an older commit? I don't think we automatically `git pull`, maybe for the sample repos we could do that.

**Comment by bamurtaugh** [comment]
> I don't think we automatically git pull, maybe for the sample repos we could do that.

I think that was my misunderstanding - I've already opened the Python repo via the Try a Sample command before, so the extension seems to be grabbing that older version. Sounds like that's expected right now, so my issue is more of a feature request than a bug 😄. 

> Is the Python repository in your Docker volume on an older commit

I looked through the Dev Containers logs as the container is set up, and I traced through hashes that look like commit numbers to try to correlate them with specific commits on Remote - Containers or remote-try-python, but I wasn't sure if I should be seeing an exact commit number referenced in the logs (or somewhere else). 

***

I just executed Try a Sample again (this time with Go) since today we [merged in changes](https://github.com/microsoft/vscode-remote-try-go/pull/30) to the Go readme and devcontainer.json. Remote-Containers remembers changes I made to this repo previously, but doesn't pull in the latest updates:

![image](https://user-images.githubusercontent.com/25310137/110399366-97481f00-802a-11eb-9cdf-ee0a118b3045.png)

If my understanding is now correct, I think doing an automatic `git pull` would be nice so that users get the latest sample setup and info. One risk I could see is that it'd overwrite local changes users may end up making (i.e. in my Go screenshot above), but I'd think 'Try a Sample' is for getting a quick/latest dev container sample, and if a user would like to maintain specific code, they'd want to "graduate" out of this flow.
